From: Mauro Carvalho Chehab <mchehab@redhat.com>
Date: Wed, 19 Nov 2008 18:26:50 -0200
Subject: [usb] add support for dell keyboard 431c:2003
Message-id: 20081119182650.57556f87@pedra.chehab.org
O-Subject: [PATCH RHEL 5.4] Add support for dell keyboard 431c:2003
Bugzilla: 471469
RH-Acked-by: Pete Zaitcev <zaitcev@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

bz471469

Bugzilla 434538 applied the fixes for dell keyboard 413c:2105. However,
dell keyboard 431c:2003 has the same bug.

This patch adds also this USB ID.

There's no upstream patch, and there's no sense on sending it upstream,
since there's a patch already at linux-next using a different approach,
That is meant to remove all keyboard led quirks, on changeset
f373bede6c601606527415bb5717a7eef90bd3fe (linux-next tree).

After having the linux-next patch upstream, it would be a good idea to
port it to RHEL, removing the hid quirks.

However, for now, it is safer to let the patch be more tested, waiting for it
to be upstream. So, let's add this as a quirk.

The patch were tested by the customer.

Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>

diff --git a/drivers/usb/input/hid-core.c b/drivers/usb/input/hid-core.c
index eeadf76..3deb026 100644
--- a/drivers/usb/input/hid-core.c
+++ b/drivers/usb/input/hid-core.c
@@ -1634,6 +1634,7 @@ static void hid_set_leds(struct hid_device *hid)
 #define USB_DEVICE_ID_RARITAN_KVM_DONGLE	0xcdab
 
 #define USB_VENDOR_ID_DELL		0x413c
+#define USB_DEVICE_ID_DELL_RT7D50	0x2003
 #define USB_DEVICE_ID_DELL_W7658	0x2005
 #define USB_DEVICE_ID_DELL_SK8115	0x2105
 
@@ -1802,6 +1803,7 @@ static const struct hid_blacklist {
 	{ USB_VENDOR_ID_PANJIT, 0x0004, HID_QUIRK_IGNORE },
 
 	{ USB_VENDOR_ID_BRIGHT, USB_DEVICE_ID_BRIGHT_ABNT2, HID_QUIRK_RESET_LEDS },
+	{ USB_VENDOR_ID_DELL, USB_DEVICE_ID_DELL_RT7D50, HID_QUIRK_RESET_LEDS },
 	{ USB_VENDOR_ID_DELL, USB_DEVICE_ID_DELL_W7658, HID_QUIRK_RESET_LEDS },
 	{ USB_VENDOR_ID_DELL, USB_DEVICE_ID_DELL_SK8115, HID_QUIRK_RESET_LEDS },
 	{ USB_VENDOR_ID_LOGITECH, USB_DEVICE_ID_LOGITECH_KBD, HID_QUIRK_RESET_LEDS },
